NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -206,6 +328,25 @@ private void initiateDefaultPerms(String group){
 		}
 	}
 	
+	public String getDefaultGroup(UUID uuid){
+		return groupManagerDao.getDefaultGroup(uuid);
+	}
+
+	/**
+	 * Invalidates a group from cache.
+	 * @param group
+	 */
+	public void invalidateCache(String group){
+		Group g = groupsByName.get(group);
+		if (g != null) {
+			g.setValid(false);
+			groupsByName.remove(group);
+			for (Group x: groupsById.values())
+				if (x.getName().equals(g.getName()))
+				groupsById.remove(x.getGroupId());
```

Comment:
Might as well break out of the for loop if the group was found and removed out of the id map.